.github/workflows/build.yml

```diff
@@ -12,6 +12,8 @@ jobs:
           go-version: '1.18'
       - name: Run tests
         run: go test -v -covermode=count -coverprofile=profile.cov ./...
+      - name: fuzz
+        run: make fuzz
```

Comment:
Adding a Makefile looks great -- what do you say about merging these tests into it? 

E.g. we could change _this_ to just be:

```yaml
- name: Run tests
  run: make test
```

And then update the Makefile to something more like this:

```yaml
.PHONY: test
test: unit fuzz

.PHONY: unit
unit:
	go test -v -covermode=count -coverprofile=profile.cov ./...

.PHONY: fuzz
fuzz:
	go test -v -cover -fuzz=Fuzz ./utils -fuzztime=3s
```

Reply:
(I _think_ `go test -fuzz=Fuzz ./...` will run all the fuzz tests?)

Reply:
go test -fuzz=Fuzz ./.. does not work because, fuzz does not work with multiple fuzz functions. I have already tried this and it gave me an error.

Reply:
I have added the rest of the changes